site/_posts/2017-08-16-quilt-arrow.md

```diff
@@ -0,0 +1,85 @@
+
+When we started, Parquet support in Python was limited. In fact, searching for an open-source library to read and write Parquet files was how we discovered the Arrow project in the first place. We tested two Parquet libraries, fastparquet and pyarrow. The two libraries performed similarly for serializing and deserializing Pandas DataFrames to and from Parquet and in early versions we included hooks for both.
+
+## Bringing Data Packages into Memory with <b>Arrow</b>
+In addition to a storage format, to present tabular data in a consistent, ready-to-analyze state across platforms, we need to define a common <b>in-memory</b> representation of tabular data in packages. Arrow is exactly that. With Arrow, tabular data can be packaged as Arrow tables, ensuring that they have the exact same presentation in Python, R and Spark.
+<img src="2017-08-15-quilt-arrow-block-diagram.svg"/>
+
+We built the  Quilt client for Python on top of Pandas and pyarrow, which in turn builds on Arrow C++ and Parquet C++. We’re currently working with the open-source community to write clients for R and Spark. Those efforts will be able to leverage the same core libraries (Arrow C++ and Parquet C++). In addition to saving code and development effort, using the same foundation libraries will ensure that the resulting Quilt clients represent data from Quilt packages in the same way.
+
+# Quilt
+[Quilt](https://github.com/quiltdata/quilt) is an open-source tool that lets users bundle data and metadata into data packages that are directly importable into Python--and soon R and Spark. The basic process includes the following steps:
+
+
+<img src="2017-08-15-quilt-arrow-operations.png" width="400" height="600" />
+
+# How it Works Today (in Python)
+<b>Build:</b>
+Quilt uses Pandas to read data from text files (e.g., CSV), databases and other sources into DataFrames. Arrow converts from DataFrames (e.g., Pandas in Python) into Arrow Tables. The Arrow Tables are then saved to Parquet using:  pyarrow.parquet.write_table.
+
+<b>Push:</b>
+Quilt users can push data packages to a [registry](https://github.com/quiltdata/quilt-registry) (like a Docker registry).
+
+<b>Install:</b>
+Install downloads package metadata and data from the registry and installs them locally.
+
+<b>Import:</b>
+Resident packages (either built locally or installed from a registry) can be imported as objects into Python (e.g., from quilt.data.uciml import iris). For tabular data nodes, Quilt uses Arrow to load data from Parquet into a ParquetDataSet. Quilt then loads the ParquetDataset into an Arrow Table using ParquetDataset.read. By default, Quilt converts the Arrow Table to a Pandas DataFrame using pyarrow.table.to_pandas(). In the future, we plan to add a method to quilt.DataNode that returns the Arrow Table so that Quilt packages can be used with Arrow-compatible analysis tools.
```

Comment:
add literal quotes to "from quilt.data.uciml import iris"

ParquetDataSet -> `ParquetDataset`

`quilt.DataNode`